core/src/com/unciv/ui/utils/KeyPressDispatcher.kt

```diff
@@ -61,14 +82,97 @@ class KeyPressDispatcher: HashMap<KeyCharAndCode, (() -> Unit)>() {
     operator fun contains(code: Int) = this.contains(KeyCharAndCode(code))
     fun remove(code: Int) = this.remove(KeyCharAndCode(code))
 
+    // access by KeyCharAndCode
+    operator fun set(key: KeyCharAndCode, action: () -> Unit) {
+        super.put(key, action)
+        checkInstall()
+    }
+    override fun remove(key: KeyCharAndCode): (() -> Unit)? {
+        val result = super.remove(key)
+        checkInstall()
+        return result
+    }
+
+    override fun toString(): String {
+        return (if (name==null) "" else "$name.") +
+            "KeyPressDispatcher(" + keys.joinToString(limit = 6){ it.toString() } + ")"
+    }
+
+    /** Removes all of the mappings, including a checkpoint if set. */
     override fun clear() {
         checkpoint = setOf()
         super.clear()
+        checkInstall()
     }
+
+    /** Set a checkpoint: The current set of keys will not be removed on a subsequent [revertToCheckPoint] */
     fun setCheckpoint() {
         checkpoint = keys.toSet()
     }
+    /** Revert to a checkpoint: Remove all mappings except those that existed on a previous [setCheckpoint] call.
+     * If no checkpoint has been set, this is equivalent to [clear] */
     fun revertToCheckPoint() {
         keys.minus(checkpoint).forEach { remove(it) }
+        checkInstall()
+    }
+
+    /** install our [EventListener] on a stage with optional inhibitor
+     * @param   stage The [Stage] to add the listener to
+     * @param   hasOpenPopups An optional lambda - when it returns true all keys are ignored
+     */
+    fun install(stage: Stage, name: String? = null, hasOpenPopups: (() -> Boolean)? = null) {
```

Comment:
please change lambda to ignoreAllKeys or something

Reply:
Ah - of course!